NEWS.rst

```diff
@@ -10,6 +10,7 @@ Removals
   and `defn` instead.
 * Literal keywords are no longer parsed differently in calls to functions
   with certain names.
+* `hy.contrib.multi` has been removed. Use ``if`` or ``cond`` instead.
```

Comment:
It's better to recommend [`functools.singledispatch`](https://docs.python.org/3.6/library/functools.html#functools.singledispatch) and/or [`multipledispatch`](https://github.com/mrocklin/multipledispatch).

Reply:
I'm not super-familiar with those, but they seem more like real multiple dispatch than the things we're replacing. `multi.defn` selects a function based on the number of arguments and `defmulti` is based on value, whereas `fuctools.singledispatch` and `multipledispatch` are based on type.  So it's hard to see how you would in general use `singledispatch` or `multipledispatch` to replace a use of `multi.defn` or `defmulti`.

Reply:
Many uses of `defn` and `defmulti` are easily covered by the type-based dispatching provided by `singledispatch` and `multipledispatch`, and&mdash;in those cases&mdash;they are much more useful recommendations than refactoring to use `if` and/or `cond`.

For instance, here's a reproduction of the `defn` example using `multipledispatch`:
```python
from multipledispatch import dispatch


def fun(a, b=None, c=None):
    return _fun(a, b, c)

@dispatch(object, type(None), type(None))
def _fun(a, b, c):
    return "a"

@dispatch(object, object, type(None))
def _fun(a, b, c):
    return "a b"

@dispatch(object, object, object)
def _fun(a, b, c):
    return "a b c"


>>> fun(1)
'a'
>>> fun(1, 2)
'a b'
>>> fun(1, 2, 3)
'a b c'
```

Reply:
Oh, neat, I didn't realize that you could do that with `multipledispatch`. What about the other cases?

Reply:
You can leave it up to the user to figure out the details for their specific problem.  Simply providing a good direction to look is more than enough.

Reply:
Okay.